### PR TITLE
Add support for Mint.UnsafeProxy connections

### PR DESCRIPTION
**Pull request by bsedat**
This adds several private proxy functions to Finch.HTTP1.Conn to dispatch to either Mint.UnsafeProxy or Mint.HTTP1. The dispatching is performed explicitly so as to better catch any changes to the underlying API with compiler errors/warnings where possible.

I don't think this completely finishes the scope of what was being discussed in #157 but it clears up the runtime errors and now handles HTTP 1.1 proxy configuration.

### CODE REVIEW

**Comment by sneako** [comment]
Thanks @bsedat, this looks good to me!

If you don't mind could you just please reorganize this module a bit? I would like to keep the public functions above the private functions.

That plus a quick `mix format` and this should be good to go.

**Comment by bsedat** [comment]
I moved the functions to a new module because they all had the same prefix, but let me know if you prefer keeping them in `Finch.Conn` and I can move them back and reorganize as requested.

File: lib/finch/http1/mint_http1.ex [hunk 1]
```diff
@@ -0,0 +1,74 @@
+defmodule Finch.MintHTTP1 do
+  @moduledoc false
+
```
Comment (sneako) [inline lib/finch/http1/mint_http1.ex:3]:
```suggestion
  @moduledoc false
  
  # This module exists to ensure that protocol negotiation does not accidentally cause Mint to open
  # HTTP2 connections in our HTTP1 pools.

```

**Comment by sneako** [comment]
Thanks @bsedat ! Let's just add this comment as a reminder as to why we have this new module, then I would say this is good to go

**Comment by sneako** [comment]
Thank you!

**Pull request merged**